src/go/util/marshal_test.go

```diff
@@ -43,28 +40,36 @@ func TestUnmarshalBytesToPbMessage(t *testing.T) {
 		}
 	}
 
-	wantResp = &confpb.Service{
-		Id: "test-id",
-	}
-	getResp = &confpb.Service{}
-	_test()
-
-	wantResp = &smpb.ListServiceRolloutsResponse{
-		NextPageToken: "next-page-token",
-	}
-	getResp = &smpb.ListServiceRolloutsResponse{}
-	_test()
-
-	wantResp = &scpb.ReportResponse{
-		ServiceConfigId: "test-id",
+	testCases := []struct {
+		getResp   proto.Message
+		wantResp  proto.Message
+		wantError string
+	}{
+		{
+			getResp: &confpb.Service{},
+			wantResp: &confpb.Service{
+				Id: "test-id",
+			},
+		},
+		{
+			getResp: &smpb.ListServiceRolloutsResponse{},
+			wantResp: &smpb.ListServiceRolloutsResponse{
+				NextPageToken: "next-page-token",
+			},
+		},
+		{
+			getResp: &scpb.ReportResponse{},
+			wantResp: &scpb.ReportResponse{
+				ServiceConfigId: "test-id",
+			},
+		},
+		{
+			getResp:   &scpb.ReportRequest{},
+			wantError: "not support unmarshalling",
+		},
 	}
-	getResp = &scpb.ReportResponse{}
-	_test()
 
-	wantResp = &scpb.ReportRequest{
-		ServiceConfigId: "test-id",
+	for _, tc := range testCases {
+		_test(tc.getResp, tc.wantResp, tc.wantError)
```

Comment:
You're passing getResp, wantResp. But func parameters are ordered wantResp, getResp